### PR TITLE
Merge structural and functional edges to build a full network

### PR DESCRIPTION
**Pull request by suganya-s**
I'm also skipping the train node_2_vec test as it's gone non-deterministic again.

In the future we should explore using the content_ids as node_ids instead of using integer node_ids

### CODE REVIEW

File: src/data_preprocessing/get_content_store_data.py [hunk 1]
```diff
@@ -238,13 +246,18 @@ def get_all_links_df(mongodb_collection, base_path_to_content_id_mapping):
...
-    base_path_to_content_id_mapping = get_base_path_to_content_id_mapping(content_store_collection)
-    # TODO: create and save down content_id_to_base_path_mapping so we can
-    #  create links for visual inspection - do they need titles?
+    page_path_content_id_mapping, content_id_base_path_mapping = get_base_path_content_id_mappings(content_store_collection)
+
+    logger.info(f'saving content_id_base_path_mapping to {data_dir}/tmp/content_id_base_path_mapping.json')
+    with open(
+            os.path.join(data_dir, 'tmp', 'base_path_to_content_id_mapping.json'),
+            'w') as content_id_base_path_file:
+        json.dump(content_id_base_path_mapping, content_id_base_path_file)
```
Comment (ellieking17) [inline src/data_preprocessing/get_content_store_data.py]: So funny, I've just done this too!

File: src/features/make_network.py [hunk 2]
```diff
@@ -35,33 +35,57 @@ def make_structural_network_and_node_id_mappings(all_links_df):
             num_counter += 1
 
     logger.info('adding source and target node_ids to all_links_df')
-    all_links_df['source'] = all_links_df['source_content_id'].map(
+    edges_df['source_node'] = edges_df['source_content_id'].map(
```
Comment (ellieking17) [inline src/features/make_network.py]: do we still need node_ids?

File: src/features/make_network.py [hunk 3]
```diff
@@ -5,19 +5,19 @@
...
     and mappings between the node_ids and content_ids
-    :param all_links_df: pandas DataFrame with columns source_content_id and destination_content_id
+    :param edges_df: pandas DataFrame with columns source_content_id and destination_content_id
     (content_ids of pages, and pages they link to)
     :return: pandas DataFrame of edges using node_ids, Python dict {content_id: node_id}, Python dict {node_id: content_id}
     """
+    # NB this function changes the existing edges_df too
     logger = logging.getLogger('make_structural_network.make_structural_network_and_node_id_mappings')
 
     content_id_node_id_mapping = {}
```
Comment (ellieking17) [inline src/features/make_network.py]: why do we need this now?

File: src/features/make_network.py [hunk 4]
```diff
@@ -35,33 +35,57 @@ def make_structural_network_and_node_id_mappings(all_links_df):
...
     logger.info(
         'creating node_id_content_id_mapping from content_id_node_id_mapping')
     node_id_content_id_mapping = dict(
         (node_id, content_id) for content_id, node_id in content_id_node_id_mapping.items())
 
-    return all_links_df, content_id_node_id_mapping, node_id_content_id_mapping
+    return edges_df, content_id_node_id_mapping, node_id_content_id_mapping
+
+
+def combine_structural_functional_edges(structural_edges, functional_edges):
```
Comment (ellieking17) [inline src/features/make_network.py]: what about weights?

File: src/models/train_node2vec_model.py [hunk 5]
```diff
@@ -11,24 +11,27 @@
...
 
-def create_graph(edges_df, node_id_content_id_mapping):
+def create_graph(edges_df
+                 # , node_id_content_id_mapping
+                 ):
     logger = logging.getLogger('train_node2_vec_model.create_graph')
     logger.info('creating graph from edges_df')
     graph = nx.convert_matrix.from_pandas_edgelist(
-        edges_df, source='source', target='target'
+        edges_df, source='source_node', target='destination_node'
```
Comment (ellieking17) [inline src/models/train_node2vec_model.py]: Without node_ids, i think this works with ```source='source_content_id, target='destination_content_id```, right?

File: src/features/make_network.py (hunk 4 above)
Comment (suganya-s) [reply src/features/make_network.py]: we're not actually using the weights anywhere so I got rid of them

File: tests/unit/data_preprocessing/test_get_content_store_data.py [hunk 6]
```diff
@@ -1448,9 +1448,11 @@ def test_convert_link_list_to_df():
 
 def test_get_base_path_to_content_id_mapping(mongodb):
     with open('tests/unit/fixtures/base_path_content_id_mapping_test_sample.json', 'r') as infile:
-        mapping = json.load(infile)
+        base_path_content_id_mapping_test_sample = json.load(infile)
```
Comment (ellieking17) [inline tests/unit/data_preprocessing/test_get_content_store_data.py]: should this be page_path? It's a little confusing but maybe the test sample truly is only base_paths not page_paths?

Comment (ellieking17) [reply tests/unit/data_preprocessing/test_get_content_store_data.py]: Can you clarify what the fixture is?

File: src/features/make_network.py [hunk 7]
```diff
@@ -67,34 +33,20 @@ def combine_structural_functional_edges(structural_edges, functional_edges):
...
 
-    all_edges_df = combine_structural_functional_edges(structural_edges_df, functional_edges_df)
-
-    all_edges_df.to_csv(os.path.join(
-        data_dir, 'tmp', 'all_edges.csv'), index=False)
-
-    module_logger.info('making network_df with node_ids using all_links')
-    network_df, content_id_node_id_mapping_dict, node_id_content_id_mapping_dict = \
-        make_network_and_node_id_mappings(all_edges_df)
+    module_logger.info('making network_df with node_ids using structural_edges_df and functional_edges_df')
```
Comment (ellieking17) [inline src/features/make_network.py]: without node_ids?

File: src/data_preprocessing/get_content_store_data.py [hunk 8]
```diff
@@ -185,17 +192,18 @@ def extract_embedded_links_df(page_text_df, base_path_to_content_id_mapping):
...
-
-    # OUTPUT_DF_COLUMNS = ['destination_base_path', 'destination_content_id', 'source_base_path', 'source_content_id']
-
-    embedded_links_df.columns = ['source_base_path', 'source_content_id', 'destination_base_path',
-                                 'destination_content_id']
+    embedded_links_df.rename(
+                             columns={
+                                 '_id': 'source_base_path',
+                                 'content_id': 'source_content_id',
+                                 'embedded_links': 'destination_base_path'},
```
[the remark this answers is not in the document]
Comment (suganya-s) [reply src/data_preprocessing/get_content_store_data.py]: it's already called `destination_content_id` so we don't have to rename it

File: src/data_preprocessing/make_functional_edges_and_weights.py [hunk 9]
```diff
@@ -42,7 +42,7 @@ def extract_df_to_csv(self, file_path):
     data_dir = os.getenv("DATA_DIR")
     blacklisted_document_types = get_excluded_document_types()
     yesterday = (datetime.today() - timedelta(1)).strftime('%Y%m%d')
-    three_weeks_ago = (datetime.today() - timedelta(22)).strftime('%Y%m%d')
+    three_weeks_ago = (datetime.today() - timedelta(2)).strftime('%Y%m%d')
```
[the remark this answers is not in the document]
Comment (suganya-s) [reply src/data_preprocessing/make_functional_edges_and_weights.py]: testing, it's back as 22 on the branch now though so not sure why you can see this

File: src/features/make_network.py (hunk 4 above)
Comment (suganya-s) [reply src/features/make_network.py]: shall we add that as a `HAVING weight >5` to the SQL?

File: tests/unit/data_preprocessing/test_get_content_store_data.py (hunk 6 above)
Comment (suganya-s) [reply tests/unit/data_preprocessing/test_get_content_store_data.py]: it's page_path, good spot

File: src/features/make_network.py (hunk 4 above)
Comment (suganya-s) [reply src/features/make_network.py]: done :)

**Pull request merged**